src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/CommandLineItemHandler.cs

```diff
@@ -33,71 +32,59 @@ public OrderPrecedenceImportCollection<ILanguageServiceCommandLineHandler> Handl
             get;
         }
 
-        public string RuleName
+        public override string RuleName
         {
             get { return CompilerCommandLineArgs.SchemaName; }
         }
 
-        public RuleHandlerType HandlerType
+        public override RuleHandlerType HandlerType
         {
             get { return RuleHandlerType.DesignTimeBuild; }
         }
 
-        public void SetContext(IWorkspaceProjectContext context)
-        {
-            Requires.NotNull(context, nameof(context));
-
-            _context = context;
+        // Broken design time builds generates updates with no changes.
+        public override bool ReceiveUpdatesWithEmptyProjectChange => true;
 
-            foreach (var handler in Handlers)
-            {
-                handler.Value.SetContext(_context);
-            }
-        }
-
-        public Task HandleAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> e, IProjectChangeDescription projectChange)
+        public override Task HandleAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> e, IProjectChangeDescription projectChange, IWorkspaceProjectContext context)
         {
             Requires.NotNull(e, nameof(e));
             Requires.NotNull(projectChange, nameof(projectChange));
 
-            if (!ProcessDesignTimeBuildFailure(projectChange))
+            if (!ProcessDesignTimeBuildFailure(projectChange, context))
             {
-                ProcessOptions(projectChange);
-                ProcessItems(projectChange);
+                ProcessOptions(projectChange, context);
+                ProcessItems(projectChange, context);
             }
 
             return Task.CompletedTask;
         }
 
-        private bool ProcessDesignTimeBuildFailure(IProjectChangeDescription projectChange)
+        private static bool ProcessDesignTimeBuildFailure(IProjectChangeDescription projectChange, IWorkspaceProjectContext context)
         {
             // WORKAROUND: https://github.com/dotnet/roslyn-project-system/issues/478
             // Check if the design-time build failed, if we have no arguments, then that is likely the 
             // case and we should ignore the results.
 
             bool designTimeBuildFailed = projectChange.After.Items.Count == 0;
-
-            _context.LastDesignTimeBuildSucceeded = designTimeBuildFailed;
-
+            context.LastDesignTimeBuildSucceeded = !designTimeBuildFailed;
```

Comment:
These are in 2 different types: command handler and LanguageServiceHost and there is not much common code.